lib/net/http_xhr_plugin.js

```diff
@@ -86,7 +98,28 @@ shaka.net.HttpXHRPlugin = class {
             shaka.util.Error.Code.TIMEOUT,
             uri, requestType));
       };
+
+      const stallTimeoutMs = request.retryParameters.stallTimeout;
+      if (stallTimeoutMs) {
+        stallTimer = new shaka.util.Timer(() => {
+          timedOut = true;
+          xhr.abort();
+          reject(new shaka.util.Error(
+              shaka.util.Error.Severity.RECOVERABLE,
+              shaka.util.Error.Category.NETWORK,
+              shaka.util.Error.Code.TIMEOUT,
+              uri, requestType));
+        });
+      }
+
       xhr.onprogress = (event) => {
+        if (connectionTimer) {
+          connectionTimer.stop();
+        }
+        if (stallTimer) {
+          stallTimer.stop();
```

Comment:
Same here.  You can drop the stop() call if you're calling tickAfter().

Reply:
Done!